site/js/MessagesActor.js

```diff
@@ -0,0 +1,425 @@
+
+      if (!this.actors.session.loggedIn) {
+        alert('You must be logged in to send a message.')
+        return
+      }
+
+      if (!this.actors.channels.activeChannelID) {
+        alert('You must be in a channel to send a message.')
+        return
+      }
+
+      const text = chatInput.value
+
+      try {
+        chatInput.value = ''
+
+        if (text.trim().length === 0) {
+          return
+        }
+
+        //const signature = await signText(text)
+        const channelID = this.actors.channels.activeChannelID
+        const sessionID = this.actors.session.sessionID
+
+        const result = await post('send-message', {
+          text,
+          //signature,
+          channelID,
+          sessionID,
+        })
+
+        if (result.success) {
+          return
+        }
+      } catch(error) {
+        console.error(error)
+      }
+
+      const restore = confirm(
+        'Your message was NOT sent! Some sort of internal error. See your browser network/console log.\n' +
+        'However, its content was saved:\n"""\n' + text + '\n"""\n' +
+        'Would you like to restore this into the chat input box?'
+      )
+
+      if (restore) {
+        chatInput.value = text
+      }
+    })
+  
+    const messagesContainer = document.getElementById('messages')
+    this.socket.on('received chat message', async msg => {
+      if (typeof msg !== 'object') {
+        return
+      }
+
+      const { revisions, authorID, id: messageID } = msg.message
+
+      if (!revisions || !authorID) {
+        return
+      }
+
+      if (!revisions.length || !revisions[0].text) {
+        return
+      }
+
+      const getScrollDist = () => messages.scrollHeight - messages.offsetHeight
+      let wasScrolledToBottom = (messages.scrollTop === getScrollDist())
+
+      const el = document.createElement('div')
+      el.classList.add('message')
+      el.setAttribute('id', 'message-' + messageID)
+      el.dataset.author = authorID
+      el.appendChild(await this.buildMessageContent(msg.message))
+      messagesContainer.appendChild(el)
+
+      if (this.actors.session.isCurrentUser(authorID)) {
+        el.classList.add('created-by-us')
+      }
+
+      if (wasScrolledToBottom) {
+        messagesContainer.scrollTop = getScrollDist()
+      }
+
+      el.addEventListener('click', async () => {
+        // Don't do anything if we don't own this message!
+        if (!actors.session.isCurrentUser(authorID)) {
+          return
+        }
+
+        const text = prompt('Edit message - new content?')
+
+        if (!text || text.trim().length === 0) {
+          return
+        }
+
+        const result = await post('edit-message', {
+          sessionID, text, messageID,
+          //signature: await signText(text)
+        })
+      })
+    })
+
+    this.socket.on('edited chat message', async msg => {
+      if (typeof msg !== 'object') {
+        return
+      }
+
+      await this.showMessageRevision(msg.message)
+    })
+  }
```

Comment:
Thanks to going to all the work to port this over. I know it's probably not that much more complicated than find-replacing `socket` with `this.socket`, but.. still! :tada: